File: sakuralearn-backend/src/main/java/com/sakuralearn/sakuralearn_backend/service/LessonService.java
package com.sakuralearn.sakuralearn_backend.service;

import com.sakuralearn.sakuralearn_backend.dto.request.LessonRequest;
import com.sakuralearn.sakuralearn_backend.dto.response.LessonResponse;
import org.springframework.web.multipart.MultipartFile;
import java.util.List;
import java.util.UUID;

public interface LessonService {
    LessonResponse createLesson(UUID courseId, LessonRequest request, MultipartFile video);
    LessonResponse updateLesson(UUID courseId, UUID lessonId, LessonRequest request, MultipartFile video);
    void deleteLesson(UUID courseId, UUID lessonId);
    List<LessonResponse> getLessonsByCourse(UUID courseId);
    void reorderLessons(UUID courseId, List<UUID> lessonIds);
}

File: sakuralearn-frontend/src/services/lessonService.js
import API from './api'

export const lessonService = {
  getLessonsByCourse: (courseId) => API.get(`/courses/${courseId}/lessons`),
  getBlocksByLesson: (lessonId) => API.get(`/lessons/${lessonId}/blocks`),
  createLesson: (courseId, formData) => API.post(`/courses/${courseId}/lessons`, formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  }),
  // Keep nested resource path to match backend contract.
  updateLesson: (courseId, id, formData) => API.put(`/courses/${courseId}/lessons/${id}`, formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  }),
  deleteLesson: (courseId, id) => API.delete(`/courses/${courseId}/lessons/${id}`),
  reorderLessons: (courseId, lessonIds) => API.post(`/courses/${courseId}/lessons/reorder`, lessonIds),
}

export const lessonBlockService = {
  getBlocksByLesson: (lessonId) => API.get(`/lessons/${lessonId}/blocks`),
  createBlock: (lessonId, data) => API.post(`/lessons/${lessonId}/blocks`, data),
  updateBlock: (lessonId, id, data) => API.put(`/lessons/${lessonId}/blocks/${id}`, data),
  deleteBlock: (lessonId, id) => API.delete(`/lessons/${lessonId}/blocks/${id}`),
  reorderBlocks: (lessonId, blockIds) => API.post(`/lessons/${lessonId}/blocks/reorder`, blockIds),
}
